stringdtype/stringdtype/src/dtype.c

```diff
@@ -162,6 +164,36 @@ stringdtype_ensure_canonical(StringDTypeObject *self)
     return self;
 }
 
+static int
+stringdtype_clear_loop(void *NPY_UNUSED(traverse_context),
+                       PyArray_Descr *NPY_UNUSED(descr), char *data,
+                       npy_intp size, npy_intp stride,
+                       NpyAuxData *NPY_UNUSED(auxdata))
+{
+    while (size--) {
+        if (data != NULL) {
+            free(*(ss **)data);
+        }
+        data += stride;
+    }
+
+    return 0;
+}
+
+static int
+stringdtype_get_clear_loop(void *NPY_UNUSED(traverse_context),
+                           PyArray_Descr *NPY_UNUSED(descr),
+                           int NPY_UNUSED(aligned),
+                           npy_intp NPY_UNUSED(fixed_stride),
+                           traverse_loop_function **out_loop,
+                           NpyAuxData **NPY_UNUSED(out_auxdata),
+                           NPY_ARRAYMETHOD_FLAGS *flags)
+{
+    *flags = NPY_METH_REQUIRES_PYAPI | NPY_METH_NO_FLOATINGPOINT_ERRORS;
```

Comment:
```suggestion
    *flags = NPY_METH_NO_FLOATINGPOINT_ERRORS;
```
You don't require the python API actually if you are u sing `free()` to free/allocate.

NumPy currently actually ignores the float flag also, because how will clearing cause that?

Reply:
Oh, I thought `NPY_METH_REQUIRES_PYAPI` was required for the clear loop, I'll try without.

Reply:
For all numpy ones, yes :).